### PR TITLE
Fix: The `_del()` and `_ndel()` operations shouldn't be able to alter cfg or cmp element

### PR DESCRIPTION
**Pull request by matthiasbeyer**
The name of this branch maybe is confusing, but follows the naming schemes
of the other PRs which do similar work.

### CODE REVIEW

File: src/libreset/avl.c [hunk 1]
```diff
@@ -163,8 +163,8 @@ static unsigned int
 delete_elements_by_predicate(
     struct avl_el** root,
     r_predf pred,
-    void* etc,
-    struct r_set_cfg* cfg
+    void const* etc,
```
Comment (matthiasbeyer) [inline src/libreset/avl.c]: This causes the compilation error.

When we define the `etc` parameter for the predicate function as `const`, we must also alter the predicate function type definition.

Is this feasible? Or should the `etc` parameter for the predicate function type be alterable?

Comment (neithernut) [inline src/libreset/avl.c]: As stated some time ago: the predicate should be able to alter the `etc` parameter.

Comment (matthiasbeyer) [inline src/libreset/avl.c]: Ok. Then I will submit fixups for this.

**Comment by neithernut** [comment]
ACK.

**Comment by matthiasbeyer** [comment]
Squashing done, waiting on CI.

**Comment by neithernut** [comment]
This may now be merged.

**Pull request merged**